index.js

```diff
@@ -306,9 +312,9 @@ Router.prototype.find = function find (method, path) {
     }
 
     var node = currentNode.find(path[0], method)
-    if (node === null) {
+    if (node === ROUTE_NOT_FOUND) {
       node = currentNode.parametricBrother
-      if (node === null) {
+      if (node === ROUTE_NOT_FOUND) {
```

Comment:
`null` is actually easier to understand here. `ROUTE_NOT_FOUND` sounds like the entire route was not found so the algorithm should return immediately, but here all it means is that it didn't match one of the node types. I'd really suggest reverting all the `ROUTE_NOT_FOUND` changes. `null` is perfectly easy to understand.